shotover-proxy/src/transforms/cassandra/sink_cluster/node_pool.rs

```diff
@@ -15,13 +16,20 @@ use tokio::sync::{watch, RwLock};
 
 #[derive(Debug)]
 pub enum GetReplicaErr {
-    NoMetadata,
+    NoPreparedMetadata,
+    NoKeyspaceMetadata,
     Other(Error),
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct KeyspaceMetadata {
+    pub replication_factor: usize,
+}
+
 #[derive(Debug)]
 pub struct NodePool {
     prepared_metadata: Arc<RwLock<HashMap<CBytesShort, PreparedMetadata>>>,
+    keyspace_metadata: Arc<RwLock<HashMap<String, KeyspaceMetadata>>>,
```

Comment:
prepared_metadata uses an `Arc<RwLock<T>>` because each connection needs to be able to mutate a global state because preparation can occur on any connection.
However that same reasoning does not apply to `keyspace_metadata`.
Each connection receives the keyspace_metadata from the topology task and has no need to share this state with other connections because they all received the same state.
Changing this type to a `HashMap<String, KeyspaceMetadata>` should be sufficient to resolve this issue.